gum/backend-linux/gumprocess-linux.c

```diff
@@ -568,7 +578,7 @@ gum_do_modify_thread (gpointer data)
   if (res == -1)
     goto failed_to_write;
 
-  res = gum_libc_ptrace (PTRACE_DETACH, ctx->thread_id, NULL, NULL);
+  res = gum_libc_ptrace (PTRACE_DETACH, ctx->thread_id, NULL, (void *) SIGCONT);
```

Comment:
Won't this swallow the signal if the process was stopped by a different signal than `SIGSTOP`?

Reply:
The `ptrace` man page says...

```
PTRACE_ATTACH
              Attach to the process specified in pid, making it a tracee
              of the calling process.  The tracee is sent a SIGSTOP, but
              will not necessarily have stopped by the completion of
              this call; use [waitpid(2)](https://man7.org/linux/man-pages/man2/waitpid.2.html)
              to wait for the tracee to stop.

              See the "Attaching and detaching" subsection for
              additional information.  (addr and data are ignored.)

              Permission to perform a PTRACE_ATTACH is governed by a
              ptrace access mode PTRACE_MODE_ATTACH_REALCREDS check; see
              below.
```

Therefore even if the tracee hasn't stopped, we have already injected SIGSTOP into the process, and so here we would send a corresponding `SIGCONT` to cause it to resume.